fix: no valid conf found issue in combiner GrubConf

JoySnow opened the pull request:
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [x] No Sensitive Data in this change?
* [x] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:

  using grubby combiner prior to grub conf files in combiner GrubConf,
  to better support the handling for rhel9 and rhel10 systems.

  For compatiable usage in GrubConf, enhanced grubby combiner with:
   - extend variables with GrubEnv data
   - add attribute version, kernel_initrds and is_kdump_iommu_enabled
   - format boot_entries to a list of BootEntry, for each, construct
   cmdline with args and root for get_grub_cmdlines search in GrubConf

resolve https://github.com/RedHatInsights/insights-core/issues/4159
RHINENG-15775

## Summary by Sourcery

Integrate the grubby combiner into the main GrubConf combiner and refactor UEFI detection to improve grub configuration parsing on RHEL9/RHEL10, resolving the "no valid conf found" issue.

New Features:
- Introduce a Grubby combiner that parses grubby --info and --default-index outputs and integrates optional GrubEnv variables
- Add IsUEFIBoot combiner to detect UEFI boot based on sysfs directory contents

Bug Fixes:
- Fix "No valid conf found" errors in GrubConf by prioritizing grubby combiner before grub config files

Enhancements:
- Refactor GrubConf combiner to accept a new Grubby combiner, unify grub variant selection, and use IsUEFIBoot for UEFI detection
- Extend BootLoaderEntries combiner to use IsUEFIBoot instead of raw ls_lan/ls_sf inputs
- Enhance Grubby combiner to generate BootEntry objects, parse raw_args, build kernel/initrd lists, and detect kdump IOMMU status lazily
- Support RHEL9 in Grub2EFIConfig parser

Tests:
- Add tests for Grubby combiner argument parsing, GrubEnv expansion, kernel/initrd extraction, and is_kdump_iommu_enabled detection
- Add RHEL9 UEFI grub2 + grubby integration tests
- Update existing combiner tests to match new GrubConf and BootLoaderEntries signatures

JoySnow commented on insights/combiners/grubby.py:
```diff
@@ -28,19 +33,119 @@
             )
+
+        self.version = self._version = 2  # GRUB2
+        self._kernel_initrds = "not set yet"  # lazy load
```
`@cached_property` is a good one, just it only support py3.8+

xiangce reviewed (commented):
Please check my comments below.

xiangce reviewed (commented):
To avoid test issue about "filter" is required for "ls_lan_filters" when using the Grubby combiner which rely on "LsLanFilter", please update the `COMPONENT_FILTERED_PARSERS`

https://github.com/RedHatInsights/insights-core/blob/2f7f3732a12a116373f426aeab220bac9cf2aef7/insights/tests/__init__.py#L160-L171

xiangce commented:
@JoySnow this PR is ready for merging, please file a new PR to `3.0_egg` branch per the steps given in [RPM/Egg Delivery](https://github.com/RedHatInsights/insights-core/blob/master/CONTRIBUTING.md#rpmegg-delivery)

JoySnow commented:
Rebased and squashed the commits for this MR. 
Raised the 3.0_egg branch backport PR at https://github.com/RedHatInsights/insights-core/pull/4575 . 
@xiangce
